app/src/main/java/bean/pwr/imskamieskiego/MapGenerator/MapDrawer.java

```diff
@@ -0,0 +1,383 @@
+    private float scaleFromDetector;
+    private float scaleFromDetectorXcenter;
+    private float scaleFromDetectorYcenter;
+    private float scaleFromDetectorMAX = 3.f;
+    private float scaleFromDetectorMIN = 0.5f;
+
+    private final int none = 0;
+    private final int drag = 1;
+    private final int zoom = 2;
+    private int mode;
+
+
+    public MapDrawer(Context context) {
+        super(context);
+        this.context = context;
+        SharedConstructing();
+    }
+
+    public MapDrawer(Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        this.context = context;
+        SharedConstructing();
+    }
+
+
+    private void SharedConstructing() {
+
+        setClickable(true);
+        originalScale = 1;
+        offsetX = 0;
+        offsetY = 0;
+        scaleFromDetector = 1.f;
+        pathPoints = new ArrayList<>();
+        mapPoints = new ArrayList<>();
+        mapPointsTypes = new Hashtable<>();
+
+        ScaleDetector = new ScaleGestureDetector(context, new ScaleListener());
+
+        setOnTouchListener(new OnTouchListener() {
+
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                ScaleDetector.onTouchEvent(event);
+
+                if (ScaleDetector.isInProgress()) return true;
+
+                PointF pointF = new PointF(event.getX(), event.getY());
+
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+
+                        pointFlast = pointF;
+                        offsetX += deltaX;
+                        offsetY += deltaY;
+                        mode = drag;
+                        break;
+
+                    case MotionEvent.ACTION_MOVE:
+
+                        if (mode == drag) {
+
+                            deltaX = pointF.x - pointFlast.x;
+                            deltaY = pointF.y - pointFlast.y;
+                            invalidate();
+
+                            if ((Math.abs(deltaX + offsetX) > originalScale * bitmapWidth * scaleFromDetector / 3)
+                                    || (Math.abs(deltaY + offsetY) > originalScale * bitmapHeight * scaleFromDetector / 3)) {
+                                deltaX = 0;
+                                deltaY = 0;
+                                return true;
+                            }
+                        }
+                        break;
+                }
+                return true;
+            }
+        });
+    }
+
+
+    /**
+     * Setting map to be drawn on canvas
+     * and getting dimensions from bitmap
+     * to set up scale parameters
+     * If floor exceeds size of
+     * floors list method gets the last one floor
+     * map possible
+     * @param floor number of the floor to be drawn
+     */
+    public void showFloor(int floor) {
+        if (floor < 0) throw new ArrayIndexOutOfBoundsException();
+        if (floor > model.getFloorCount()) {
```

Comment:
Większe równe, a nie samo większe. Jeśli pięter mamy np. 5, to numer ostatniego piętra to 4, bo jest liczone od 0.